spec/jobs/record_payment_job_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+require "geckoboard"
+
+RSpec.describe RecordPaymentJob do
+  let(:claim) { build(:claim) }
+  let(:payment) { build(:payment, :with_figures, updated_at: DateTime.now, claim: claim) }
+
+  subject { described_class.new }
+
+  it "sends the claim reference, policy and payment date to claims.submitted.ENV dataset" do
+    ClimateControl.modify ENVIRONMENT_NAME: "environment_name" do
```

Comment:
In some other specs, instead of fixing the `ENVIRONMENT_NAME` variable, you just check that the `claims.submitted.test` dataset is being used. I'm curious why you chose a different approach here?

Reply:
With this one, we're stubbing out the env to make sure that it can be changed, and there's not some side-effect somewhere else that's hardcoding the dataset name to have `test` in there. It's also trickier to stub the ENV in the feature specs, and we don't need to do it, as we've already got the confidence here.